mozdownload/scraper.py

```diff
@@ -776,6 +782,10 @@ def __init__(self, branch='mozilla-central', build_number=None, date=None,
     def get_build_info(self):
         "Defines additional build information"
 
+        # Find branch by application
+        if self.branch is None:
+            self.branch = APPLICATIONS[self.application]
```

Comment:
Well, you are adding this code here to the `DailyScraper` class, but it really has to be part of the `Scraper` class to be available for each and every scraper. Do this e.g after line 130 when `self.application` gets set.